app/api/hooks.ts

```diff
@@ -49,14 +49,10 @@ const handleResult =
   <Data>(result: ApiResult<Data>) => {
     if (result.type === 'success') return result.data
 
-    // if logged out, hit /login to trigger login redirect
-    // Exception: 401 on password login POST needs to be handled in-page
+    // if logged out, show session expired popup and prompt to log in again
+    // Exception: 401 on password login POST needs to be handled in-form
     if (result.response.status === 401 && method !== 'loginLocal') {
-      // TODO-usability: for background requests, a redirect to login without
-      // warning could come as a surprise to the user, especially because
-      // sometimes background requests are not directly triggered by a user
-      // action, e.g., polling or refetching when window regains focus
-      navToLogin({ includeCurrent: true })
+      setSessionExpired()
```

Comment:
If we also do this here

```ts
onlineManager.setOnline(false)
```

https://tanstack.com/query/v5/docs/framework/react/guides/network-mode
https://tanstack.com/query/latest/docs/reference/onlineManager

that does work to prevent subsequent requests from going through, and therefore to prevent subsequent navs that require an API call, but it does that by making any queries and mutations sit in pending forever. Submitting a form starts a spinner that never stops, and trying to nav kicks of a loading bar that never completes. This is a bizarre experience.

Maybe this is an argument for putting some kind of modal background on the toast that prevents you from clicking anything else on screen. But that would prevent you from scrolling the page (unless we got creative and allowed scroll but not click — a very strange thing to do), which would be useful if you had filled out a longer form like instance create or firewall rule and wanted to note down what you had done.